### ISSUE TITLE
Unable to disable annotation rule for annotations with arguments

### ISSUE THREAD

**Issue by theoriginalbit** [realm/SwiftLint#3316]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Given the following in the config
```swift
attributes:
  always_on_same_line: ["@IBAction", "@NSManaged", "@objc", "@objcMembers"]
```

The code below has a warning stating it should be on its own line
```swift
@objc(EMGridMetrics) class GridMetrics: NSObject {
  // …
}
```

```bash
$ swiftlint lint

…/GridMetrics.swift:1:22: warning: Attributes Violation: Attributes should be on their own lines in functions and types, but on the same line as variables and imports. (attributes)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? **0.40.0**
* Installation method used (Homebrew, CocoaPods, building from source, etc)? **Homebrew**
* Paste your configuration file:

```yml
included:
  - Sources
opt_in_rules:
  - attributes

attributes:
  always_on_same_line: ["@IBAction", "@NSManaged", "@objc", "@objcMembers"]
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 11, and Xcode 12
* Do you have a sample that shows the issue? See above

**Comment by jpsim** [comment]
Fixing in #3405
